test/realtime/vehicle_test.exs

```diff
@@ -682,7 +682,7 @@ defmodule Realtime.VehicleTest do
       assert encoded_string =~ "\"route_id\":\"28\""
 
       assert encoded_string =~
-               "\"data_discrepancies\":[{\"attribute\":\"trip_id\",\"sources\":[{\"id\":\"swiftly\",\"value\":\"swiftly-trip-id\"},{\"id\":\"busloc\",\"value\":\"busloc-trip-id\"}]},{\"attribute\":\"route_id\",\"sources\":[{\"id\":\"swiftly\",\"value\":null},{\"id\":\"busloc\",\"value\":\"busloc-route-id\"}]}]"
+               "\"data_discrepancies\":[{\"sources\":[{\"id\":\"swiftly\",\"value\":\"swiftly-trip-id\"},{\"id\":\"busloc\",\"value\":\"busloc-trip-id\"}],\"attribute\":\"trip_id\"},{\"sources\":[{\"id\":\"swiftly\",\"value\":null},{\"id\":\"busloc\",\"value\":\"busloc-route-id\"}],\"attribute\":\"route_id\"}]"
```

Comment:
Maybe this should be removed and be replaced with something like this?
```
assert vehicle == (vehicle |> Jason.encode!() |> Jason.decode!())
```

Reply:
We could even reduce the scope down to just the `data_discrepancies` field to be closer to this assertion.

Reply:
Yeah you might also need to deal with string vs. atom keys but overall I like both of those ideas, maybe you could also integrate the other couple of assertions in this test somehow? Like make a `decoded_vehicle` by decoding `encoded_string` and then asserting things about specific fields?

Reply:
PTALA 35b21a404b8e78099b3f4226ce3a27b7c0d8a878 🙏🏻